opal/datatype/opal_convertor.c

```diff
@@ -49,19 +49,6 @@
 extern int opal_convertor_create_stack_with_pos_general( opal_convertor_t* convertor,
                                                          int starting_point, const int* sizes );
 
-static void opal_convertor_construct( opal_convertor_t* convertor )
-{
-    convertor->pStack         = convertor->static_stack;
-    convertor->stack_size     = DT_STATIC_STACK_SIZE;
-    convertor->partial_length = 0;
-    convertor->remoteArch     = opal_local_arch;
-    convertor->flags          = OPAL_DATATYPE_FLAG_NO_GAPS | CONVERTOR_COMPLETED;
-#if OPAL_CUDA_SUPPORT
-    convertor->cbmemcpy       = &opal_cuda_memcpy;
-#endif
-}
-
-
```

Comment:
The convertor constructor and destructor must remain static. You get access to them by calling OBJ_CONSTRUCT  or OBJ_DESTRUCT, respectively.

This patch does the right thing, it should however be amended to remove all changes to the opal_convetor.[ch] files and use ```OBJ_CONSTRUCT(&convertor, opal_convertor_t)``` instead of ```opal_convertor_construct(&convertor)``` and ```OBJ_DESTRUCT(&convertor)``` instead of ```opal_convertor_cleanup(&convertor)```.